.pre-commit-hooks.yaml

```diff
@@ -0,0 +1,8 @@
+- id: cmake-format
+  name: cmake-format
+  description: cmake-format can format your listfiles nicely so that they don't look like crap.
+  entry: cmake-format
+  language: python
+  types: [file, text]
+  files: '(CMakeLists\.txt|\.cmake|\.cmake\.in)$'
```

Comment:
I don't think you want to include `.in` files since `cmake-format` will choke on `@VARIABLE@` replacements.

Reply:
We wrap `.in` file formatting using [this pattern](https://gitlab.kitware.com/utils/source-formatters/blob/master/src/c/format.clang-format-kwsys). This is used inside a larger formatting infrastructure, but the `sed` replacements are the important bit.

Reply:
I would suggest instead:

```yaml
    types: [cmake]
```

(no `files`)

Reply:
OK, thanks @asottile, I've done this.

Reply:
Hi, after changing the type to [cmake] in the .pre-commit-hooks.yaml file, the pre-commit just couldn't find the files. Is the cmake type support for the latest pre-commit?

Reply:
Yes, you need a recent version of the 'identify' library, as CMake support was only added in August, I think. I expect if you get the latest version of pre-commit it will depend on a new-enough version of 'identify'.